crates/task-impls/src/quorum_vote/handlers.rs

```diff
@@ -205,11 +200,23 @@ pub(crate) async fn handle_quorum_proposal_validated<
                             "Failed to convert the serialized QC signature into a DRB seed input."
                         );
                     };
-                    let new_drb_task =
-                        spawn(async move { compute_drb_result::<TYPES>(drb_seed_input) });
+
+                    // Store the drb seed input for the next calculation
+                    task_state
+                        .drb_computations
+                        .store_seed(new_epoch_number, drb_seed_input);
+                }
+
+                // Start the new task if we're in the committee for this epoch
+                if task_state
+                    .membership
+                    .has_stake(&task_state.public_key, current_epoch_number)
+                {
+                    // This will join_or_abort_task before starting the new task
                     task_state
                         .drb_computations
-                        .insert(new_epoch_number, new_drb_task);
+                        .start_new_task(current_epoch_number)
+                        .await;
```

Comment:
Similar to my comment above, this part should be at the start of this function to make it unrelated to the `Decide` status.